File: src/main/java/src/Calendario/Beca.java
package Calendario;

import java.util.ArrayList;
import java.util.Collections;
import java.util.Comparator;

import personas.Estudiante;

public class Beca {

	
	private String nombre;
	private tipos tipo;
	private enum tipos {becaInicial, becaNormal, becaAvanzada};
	public static ArrayList<Estudiante> estudiantes;
	protected ArrayList<Estudiante> estudiantesAptosInicial;
	protected ArrayList<Estudiante> estudiantesAptosNormal;
	protected ArrayList<Estudiante> estudiantesAptosAvanzada;
	
	
	public Beca(String nombre) {
		this.nombre = nombre;
		estudiantes = new ArrayList<Estudiante>();
		estudiantesAptosInicial = new ArrayList<Estudiante>();
		estudiantesAptosNormal = new ArrayList<Estudiante>();
		estudiantesAptosAvanzada = new ArrayList<Estudiante>();
	}
	
	
	public void asignarEstudiantesBeca() {
		estudiantesAptosInicial = new ArrayList<Estudiante>();
		estudiantesAptosNormal = new ArrayList<Estudiante>();
		estudiantesAptosAvanzada = new ArrayList<Estudiante>();
	        
		for (Estudiante estudiante : estudiantes) {
	            if (estudiante.getPorcentajeDeAvance() >= 20 && estudiante.getPorcentajeDeAvance() < 40 &&
	            		estudiante.calcularPromedio() >= 4.5 &&
	            		estudiante.getFueBecado() == false) {
	            	estudiantesAptosInicial.add(estudiante);
	            	}
	            
	            else if (estudiante.getPorcentajeDeAvance() >= 40 && estudiante.getPorcentajeDeAvance() < 60 &&
	            		estudiante.calcularPromedio() >= 4.0 &&
	            		estudiante.getFueBecado() == false) {
	            	estudiantesAptosNormal.add(estudiante);
			    	
				}
			    else {
			    	estudiantesAptosAvanzada.add(estudiante);
			    }
	        }
	        Collections.sort(estudiantesAptosInicial, Comparator.comparingDouble(Estudiante::calcularPromedio).reversed());
	        Collections.sort(estudiantesAptosNormal, Comparator.comparingDouble(Estudiante::calcularPromedio).reversed());
	        Collections.sort(estudiantesAptosAvanzada, Comparator.comparingDouble(Estudiante::calcularPromedio).reversed());
	        
	        for (int i = 0; i<2;i++) {
	        	this.estudiantesAptosInicial.add(estudiantesAptosInicial.get(i));
	        	this.estudiantesAptosNormal.add(estudiantesAptosNormal.get(i));
	        	this.estudiantesAptosAvanzada.add(estudiantesAptosAvanzada.get(i));
	        }
		}
}
	

File: src python/gestorAplicacion.Calendario/Beca.py
import pickle

class Beca:
    estudiantes = []
    estudiantesAptosInicial = []
    estudiantesAptosNormal = []
    estudiantesAptosAvanzada = []
    
    def __init__(self, nombre):
        self.nombre = nombre
    
    def getEstudiantes(self):
        return self.estudiantes
    
    def setEstudiantes(self, estudiantes):
        Beca.estudiantes = estudiantes
    
    def asignarEstudiantesBeca(self):
        estudiantesAptosInicial2 = []
        estudiantesAptosNormal2 = []
        estudiantesAptosAvanzada2 = []
        for estudiante in self.estudiantes:
            if estudiante.getPorcentajeDeAvance() >= 20 and estudiante.getPorcentajeDeAvance() < 40 and estudiante.calcularPromedio() >= 4.5 and estudiante.getFueBecado() == False:
                estudiantesAptosInicial2.append(estudiante)
            elif estudiante.getPorcentajeDeAvance() >= 40 and estudiante.getPorcentajeDeAvance() < 60 and estudiante.calcularPromedio() >= 4.0 and estudiante.getFueBecado() == False:
                estudiantesAptosNormal2.append(estudiante)
            elif estudiante.getPorcentajeDeAvance() >= 60 and estudiante.getPorcentajeDeAvance() < 100 and estudiante.calcularPromedio() >= 3.5 and estudiante.getFueBecado() == False:
                estudiantesAptosAvanzada2.append(estudiante)
        estudiantesAptosInicial2.sort(key=lambda x: x.calcularPromedio(), reverse=True)
        estudiantesAptosNormal2.sort(key=lambda x: x.calcularPromedio(), reverse=True)
        estudiantesAptosAvanzada2.sort(key=lambda x: x.calcularPromedio(), reverse=True)
        if len(estudiantesAptosInicial2) >= 2:
            for i in range(2):
                self.estudiantesAptosInicial.append(estudiantesAptosInicial2[i])
        elif len(estudiantesAptosInicial2) == 1:
            self.estudiantesAptosInicial.append(estudiantesAptosInicial2[0])
        if len(estudiantesAptosNormal2) >= 2:
            for i in range(2):
                self.estudiantesAptosNormal.append(estudiantesAptosNormal2[i])
        elif len(estudiantesAptosNormal2) == 1:
            self.estudiantesAptosNormal.append(estudiantesAptosNormal2[0])
        if len(estudiantesAptosAvanzada2) >= 2:
            for i in range(2):
                self.estudiantesAptosAvanzada.append(estudiantesAptosAvanzada2[i])
        elif len(estudiantesAptosAvanzada2) == 1:
            self.estudiantesAptosAvanzada.append(estudiantesAptosAvanzada2[0])
    
    def getEstudiantesAptosInicial(self):
        return self.estudiantesAptosInicial
    
    def getEstudiantesAptosNormal(self):
        return self.estudiantesAptosNormal
    
    def getEstudiantesAptosAvanzada(self):
        return self.estudiantesAptosAvanzada
